### PR TITLE
Leverage wormscan client on sourcetx middleware

### PR DESCRIPTION
**Pull request by mat1asm**

### CODE REVIEW

File: relayer/application.ts [hunk 1]
```diff
@@ -56,10 +56,10 @@ export type FetchaVaasOpts = {
 };
 
 export const defaultWormholeRpcs = {
-  [Environment.MAINNET]: ["https://api.wormscan.io"],
+  [Environment.MAINNET]: ["https://api.wormholescan.io"],
   [Environment.TESTNET]: [
     "https://wormhole-v2-testnet-api.certus.one",
-    "https://api.testnet.wormscan.io",
+    "https://api.testnet.wormholescan.io",
```
Comment (mat1asm) [inline relayer/application.ts:62]: changed because rate limiting in  testnet.wormholescan.io is 1000/min vs wormscan that is 100/min

Comment (iturricf) [reply relayer/application.ts:62]: does this apply for mainnet as well?

**Review by iturricf — APPROVED** [review]

File: relayer/application.ts (hunk 1 above)
Comment (mat1asm) [reply relayer/application.ts:62]: No, both domains have the same limit: 1000, but figured we should just update it

File: relayer/middleware/source-tx.middleware.ts [hunk 2]
```diff
@@ -106,41 +127,20 @@ export async function fetchVaaHash(
...
-    } catch (error) {
-      const httpBodyText =
-        body !== undefined ? `\nHTTP response body: ${body}` : "";
-      const errorMessage = `could not obtain txHash, attempt: ${attempt} of ${retries}.${httpBodyText}
-Error: ${error}`;
-      logger?.error(errorMessage);
-      await sleep((attempt + 1) * 200); // linear wait
-    }
-  } while (++attempt < retries && !txHash);
+  let txHash = response.data?.txHash || "";
```
Comment (abhidtu2014) [inline relayer/middleware/source-tx.middleware.ts:143]: May be we can throw an error if we don't find `txHash` on the response data object and remove empty string as fallback.

File: relayer/rpc/wormscan-client.ts [hunk 3]
```diff
@@ -35,8 +35,35 @@ export class WormscanClient {
...
+    chain: number,
+    emitterAddress: string,
+    sequence: bigint,
+    opts?: WormscanOptions,
+  ): Promise<WormscanResult<WormscanVaa>> {
+    try {
+      const response = await this.client.get<{ data: WormscanVaa }>(
+        `${
+          this.baseUrl
+        }api/v1/vaas/${chain}/${emitterAddress}/${sequence.toString()}`,
```
Comment (abhidtu2014) [inline relayer/rpc/wormscan-client.ts:52]: Are we missing a backslash here `/` after `this.baseUrl`?
```suggestion
        }/api/v1/vaas/${chain}/${emitterAddress}/${sequence.toString()}`,
```

**Review by abhidtu2014 — CHANGES_REQUESTED** [review]
Added some comments, PTAL

File: relayer/middleware/source-tx.middleware.ts (hunk 2 above)
Comment (solanoepalacio) [reply relayer/middleware/source-tx.middleware.ts:143]: Maybe we should put a comment somewhere (or ideally on documentation if we had any), but...
It's a design decision not to throw an error. This is becase you wan't to leave this decision to the user of relayer-engine. The source tx hash is strictly necessary for processing some VAAs, but not for others, so it's up to the user to decide.
At some point we thought that maybe we can make this a configuration of the middleware but never got to implement it... for the time been I'd leave it as is

**Review by solanoepalacio — APPROVED** [review]

**Comment by solanoepalacio** [comment]
Appreciate that you went the extra mile and implemented this :black_heart:

File: relayer/rpc/wormscan-client.ts (hunk 3 above)
Comment (mat1asm) [reply relayer/rpc/wormscan-client.ts:52]: good question.. its not needed, given `this.baseUrl` is an instance of URL, that returns the trailing `/ ` when calling toString()

Comment (abhidtu2014) [reply relayer/rpc/wormscan-client.ts:52]: Got it, thanks 👍

File: relayer/middleware/source-tx.middleware.ts [hunk 4]
```diff
@@ -57,13 +69,23 @@ export function sourceTx(
...
+  let wormscan: WormscanClient;
   const alreadyFetchedHashes = new LRUCache({ max: 1_000 });
 
   return async (ctx, next) => {
     if (!opts) {
       // initialize options now that we know the environment from context
       opts = Object.assign({}, defaultOptsByEnv[ctx.env], optsWithoutDefaults);
     }
+    if (!wormscan) {
+      wormscan = new WormscanClient(new URL(opts.wormscanEndpoint), {
```
Comment (gabzim) [inline relayer/middleware/source-tx.middleware.ts:81]: We're trying to deprecate the name `wormscan` (hence the URL change), could we change all references to `wormscan` to `wormholescan` instead?

**Review by gabzim — CHANGES_REQUESTED** [review]

File: relayer/middleware/source-tx.middleware.ts (hunk 4 above)
Comment (mat1asm) [reply relayer/middleware/source-tx.middleware.ts:81]: makes sense

Comment (mat1asm) [reply relayer/middleware/source-tx.middleware.ts]: changed

**Review by abhidtu2014 — APPROVED** [review]

**Review by solanoepalacio — APPROVED** [review]

**Review by gabzim — APPROVED** [review]

**Pull request merged**